src/segments/sitecore.go

```diff
@@ -0,0 +1,96 @@
+package segments
+
+import (
+	"encoding/json"
+	"path/filepath"
+	"strings"
+
+	"github.com/jandedobbeleer/oh-my-posh/src/platform"
+	"github.com/jandedobbeleer/oh-my-posh/src/properties"
+)
+
+type Sitecore struct {
+	props properties.Properties
+	env   platform.Environment
+
+	EnvironmentName    string
+	IsCloudEnvironment bool
+	EnvironmentIcon    string
+}
+
+type SitecoreConfig struct {
+	Endpoints struct {
+		Default struct {
+			Ref        string `json:"ref"`
+			AllowWrite bool   `json:"allowWrite"`
+			Host       string `json:"host"`
+			Variables  struct {
+			} `json:"variables"`
+		} `json:"default"`
+	} `json:"endpoints"`
+}
+
+const (
+	CloudIcon properties.Property = "cloud_icon"
+	OtherIcon properties.Property = "other_icon"
+)
+
+func (s *Sitecore) Enabled() bool {
+	return s.shouldDisplay()
+}
+
+func (s *Sitecore) Template() string {
+	return " {{ if .PackageManagerIcon }} {{ .PackageManagerIcon }} {{ end }} {{ .EnvironmentName }} "
+}
+
+func (s *Sitecore) Init(props properties.Properties, env platform.Environment) {
+	s.props = props
+	s.env = env
+}
+
+func (s *Sitecore) shouldDisplay() bool {
+	sitecoreDir, err := s.env.HasParentFilePath(".sitecore")
+	if err != nil {
+		return false
+	}
+
+	if !sitecoreDir.IsDir {
+		return false
+	}
+
+	if !s.env.HasFilesInDir(sitecoreDir.Path, "user.json") {
+		return false
+	}
+
+	sitecoreConfigFile := filepath.Join(sitecoreDir.Path, "user.json")
+
+	if len(sitecoreConfigFile) == 0 {
+		return false
+	}
+
+	content := s.FileContentWithoutBom(sitecoreConfigFile)
+
+	var config SitecoreConfig
+	if err := json.Unmarshal([]byte(content), &config); err != nil {
+		return false
+	}
+
+	// sitecore xm cloud always has sitecorecloud.io as domain
+	s.IsCloudEnvironment = strings.Contains(config.Endpoints.Default.Host, "sitecorecloud.io")
+
+	if s.IsCloudEnvironment {
+		s.EnvironmentIcon = s.props.GetString(CloudIcon, "\uf65e")
+	} else if !s.IsCloudEnvironment {
+		s.EnvironmentIcon = s.props.GetString(OtherIcon, "\uf98a")
+	}
+
+	s.EnvironmentName = config.Endpoints.Default.Host
+
+	return true
+}
+
+func (s *Sitecore) FileContentWithoutBom(file string) string {
```

Comment:
this function also exists in the `azure` segment, can this become a global function instead?

Reply:
It could become global. Will you make this global or should I place it somewhere central? As I literally don't know anything about the main project structure (and I am new to go), I could use some guidance if you would let me do this.

Reply:
You can actually make the function in the azure segment global by removing the azure struct "attachment". By default functions are usable inside the module unless bound to a struct.

Reply:
I removed it, as I decided it was unnecessary to have this function. sorry for using to multiple commits, but I forgot to remove the commented function. I read about rebasing/rewording the commits, but I am not sure if i was allowed to force push the changes etc.

Reply:
You can force push all you want, but don't worry. I'm already doing it.